File: iPhone/VerBank-CSTSv3-IOSClientApi/ClientStation/api/trade/TradeResult_MktCFD.m
//
//  TradeResult_MktCFD.m
//  VerBank-IOSClientStation
//
//  Created by Allone on 15/7/29.
//  Copyright (c) 2015年 zhanglei. All rights reserved.
//

#import "TradeResult_MktCFD.h"
#import "TradeServiceErrCodeTable.h"



@interface TradeResult_MktCFD(){
    NSString *_errCode;
}

@end

@implementation TradeResult_MktCFD

@synthesize _errMessage;
@synthesize result;
@synthesize newMktPrice;
@synthesize orderHis;

- (id)init{
    if (self = [super init]) {
        _errCode = @"";
        _errMessage = @"";
    }
    return self;
}

- (NSString *)getErrCode{
    return _errCode;
}

- (void)setErrCodeAndCreateResult:(NSString *)code{
    _errCode = code;
    if (code == nil) {
        return;
    } else if ([code isEqualToString:TS_ERR_MD5DigestDontMatch]){
        result = RESULT_MD5DigestDontMatch;
    } else if ([code isEqualToString:TS_ERR_TradeDisabled]){
        result = RESULT_TradeDisabled;
    } else if ([code isEqualToString:TS_ERR_System_is_Close]){
        result = RESULT_System_is_Close;
    } else if ([code isEqualToString:TS_ERR_Price]){
        result = RESULT_PriceErr;
    } else if ([code isEqualToString:TS_ERR_Amount_Match]){
        result = RESULT_AmountErr;
    } else if ([code isEqualToString:TS_ERR_Cut_Loss_User]){
        result = RESULT_OutofNOPL;
    } else if ([code isEqualToString:TS_ERR_Cut_Loss_Account]){
        result = RESULT_OutofNOPL;
    } else if ([code isEqualToString:TS_ERR_Amount_limit_by_USD]){
        result = RESULT_TooManyPositions;
    } else if ([code isEqualToString:TS_ERR_Amount_Max_by_USD]){
        result = RESULT_TooManyPositions;
    } else if ([code isEqualToString:ERR_PreSellErr]){
        result = RESULT_PreSellErr;
    } else if ([code isEqualToString:TS_ERR_MarginNotEnough_Trade]){
        result = RESULT_MarginIsNotEnoughToTrade;
    } else if ([code isEqualToString:TS_ERR_MarginNotEnough_UnLock]){
        result = RESULT_MarginIsNotEnoughToTrade;
    } else if ([code isEqualToString:TS_ERR_Uptrade]){
        result = RESULT_UPTradeFailed;
    } else if ([code isEqualToString:ERR_SIGNATURE_CHECK_FAILURE]){
        result = RESULT_SIGNATURE_CHECK_FAILURE;
    } else {
        result = RESULT_otherErr;
    }
}

@end

File: WeChatService/EX02-CSTSv3-ClientAPI/src/jedi/ex01/client/station/api/trade/TradeResult_MktCFD.java
package jedi.ex01.client.station.api.trade;

import jedi.ex01.CSTS3.comm.struct.TOrderHis4CFD;
import jedi.ex01.CSTS3.proxy.comm.IPOPErrCodeTable;

public class TradeResult_MktCFD {
	public final static int RESULT_SUCCEED = 0;
	public final static int RESULT_PRICECHANGED = 1;
	public final static int RESULT_MD5DigestDontMatch = -1;
	public final static int RESULT_TradeDisabled = -2;
	public final static int RESULT_PriceErr = -3;
	public final static int RESULT_LotsErr = -4;
	public final static int RESULT_OutofNOPL = -5;
	public final static int RESULT_TooManyPositions = -6;
	public final static int RESULT_PreSellErr = -7;
	public final static int RESULT_MarginIsNotEnoughToTrade = -8;
	public final static int RESULT_UPTradeFailed = -9;
	public final static int RESULT_System_is_Close = -10;
	public final static int RESULT_otherErr = -100;

	private int result;
	private String _errCode = "";
	private String _errMessage = "";
	private double newMktPrice;
	private TOrderHis4CFD orderHis;

	public int getResult() {
		return result;
	}

	public void setResult(int result) {
		this.result = result;
	}

	public String get_errCode() {
		return _errCode;
	}

	public String get_errMessage() {
		return _errMessage;
	}

	public void set_errMessage(String message) {
		_errMessage = message;
	}

	public double getNewMktPrice() {
		return newMktPrice;
	}

	public void setNewMktPrice(double newMktPrice) {
		this.newMktPrice = newMktPrice;
	}

	public void setErrCodeAndCreateResult(String code) {
		_errCode = code;
		if (code == null) {
			return;
		} else if (code.equals(IPOPErrCodeTable.ERR_MD5DigestDontMatch)) {
			result = RESULT_MD5DigestDontMatch;
		} else if (code.equals(IPOPErrCodeTable.ERR_TradeDisabled)) {
			result = RESULT_TradeDisabled;
		} else if (code.equals(IPOPErrCodeTable.ERR_System_is_Close)) {
			result = RESULT_System_is_Close;
		} else if (code.equals(IPOPErrCodeTable.ERR_PriceErr)) {
			result = RESULT_PriceErr;
		} else if (code.equals(IPOPErrCodeTable.ERR_LotsErr)) {
			result = RESULT_LotsErr;
		} else if (code.equals(IPOPErrCodeTable.ERR_OutofNOPL)) {
			result = RESULT_OutofNOPL;
		} else if (code.equals(IPOPErrCodeTable.ERR_TooManyPositions)) {
			result = RESULT_TooManyPositions;
		} else if (code.equals(IPOPErrCodeTable.ERR_PreSellErr)) {
			result = RESULT_PreSellErr;
		} else if (code
				.equals(IPOPErrCodeTable.ERR_MarginIsNotEnoughToTrade)) {
			result = RESULT_MarginIsNotEnoughToTrade;
		} else if (code.equals(IPOPErrCodeTable.ERR_UPTRADEFAILED)) {
			result = RESULT_UPTradeFailed;
		} else {
			result = RESULT_otherErr;
		}
	}

	public TOrderHis4CFD getOrderHis() {
		return orderHis;
	}

	public void setOrderHis(TOrderHis4CFD orderHis) {
		this.orderHis = orderHis;
	}
}
